ext/gd/gd.c

```diff
@@ -4221,29 +4222,29 @@ static void _php_image_output_ctx(INTERNAL_FUNCTION_PARAMETERS, int image_type,
 
 	switch(image_type) {
 		case PHP_GDIMG_TYPE_JPG:
-			(*func_p)(im, ctx, q);
+			(*func_p)(im, ctx, quality);
 			break;
 		case PHP_GDIMG_TYPE_WEBP:
-			if (q == -1) {
-				q = 80;
+			if (quality == -1) {
```

Comment:
Could we use 80 as the default param value instead? Of course that would need some adjustment to the `quality` variable's default value.

Reply:
No, because for JPEG and PNG the default has to be -1.

Reply:
Yes, although we could use a different variable for this purpose, or adapt `qualitiy`'s default value accordingly in case of `PHP_GDIMG_TYPE_WEBP`.

Reply:
Hmm, I wonder if we're not better off to drop `_php_image_output_ctx()` altogether (i.e. inline the function, and extract the stream and ctx code to separate functions).

Reply:
Yes, I'd definitely be in favour, since `_php_image_output_ctx()` in its current state is just a big mess. Should this be made in this PR? TBH I'd prefer not to do extra work in this one so that unbundling of xml-rpc gets "unblocked".

Reply:
TBH I'd prefer if this function got split up first, and then this PR rebased on top, because all the cases are really hard to understand right now :(